key/key.go

```diff
@@ -86,7 +90,15 @@ func (k Key256) CommonPrefixLength(o Key256) int {
 
 // Compare compares the numeric value of the key with another key of the same type.
 func (k Key256) Compare(o Key256) int {
-	return bytes.Compare(k.b[:], o.b[:])
+	if k.b != nil && o.b != nil {
```

Comment:
could it actually happen that a user of Key256 ends up with a nil `.b` field? I'm just wondering and all good with being defensive here.

Reply:
`var k key.Key256` will do it